core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCatalog.java

```diff
@@ -589,4 +593,31 @@ public void testTablePropsDefinedAtCatalogLevel() throws IOException {
         "table-key5",
         table.properties().get("key5"));
   }
+
+  @Test
+  public void testRegisterTable() throws IOException {
+    TableIdentifier identifier = TableIdentifier.of("a", "t1");
+    TableIdentifier identifier2 = TableIdentifier.of("a", "t2");
+    HadoopCatalog catalog = hadoopCatalog();
+    catalog.createTable(identifier, SCHEMA);
+    Table registeringTable = catalog.loadTable(identifier);
+    TableOperations ops = ((HasTableOperations) registeringTable).operations();
+    String metadataLocation = ((HadoopTableOperations) ops).current().metadataFileLocation();
+    Assertions.assertThat(catalog.registerTable(identifier2, metadataLocation)).isNotNull();
+    Table newTable = catalog.loadTable(identifier2);
```

Comment:
nit: no need for the variable if it is used only once. Can be

`Assertions.assertThat(catalog.loadTable(identifier2)).isNotNull();`